web/src/main/kotlin/no/nav/su/se/bakover/web/routes/søknadsbehandling/LeggTilGrunnlagBosituasjonRoutes.kt

```diff
@@ -137,3 +148,43 @@ internal fun Route.leggTilGrunnlagBosituasjonRoutes(
         }
     }
 }
+
+internal fun SøknadsbehandlingService.KunneIkkeVilkårsvurdere.tilResultat(): Resultat {
+    return when (this) {
+        SøknadsbehandlingService.KunneIkkeVilkårsvurdere.FantIkkeBehandling -> fantIkkeBehandling
+        is SøknadsbehandlingService.KunneIkkeVilkårsvurdere.FeilVedValideringAvBehandlingsinformasjon -> this.feil.tilResultat()
+        SøknadsbehandlingService.KunneIkkeVilkårsvurdere.HarIkkeEktefelle -> harIkkeEktefelle
+    }
+}
+
+internal fun Søknadsbehandling.KunneIkkeOppdatereBosituasjon.tilResultat(): Resultat {
+    return when (this) {
+        is Søknadsbehandling.KunneIkkeOppdatereBosituasjon.KlarteIkkeHentePersonIPdl -> Feilresponser.fantIkkePerson
+        is Søknadsbehandling.KunneIkkeOppdatereBosituasjon.KunneIkkeLageGrunnlagsdata -> this.feil.tilResultat()
+        is Søknadsbehandling.KunneIkkeOppdatereBosituasjon.UgyldigTilstand -> ugyldigTilstand(this.fra, this.til)
+    }
+}
+
+internal fun KunneIkkeLageGrunnlagsdata.tilResultat(): Resultat {
+    return when (this) {
+        KunneIkkeLageGrunnlagsdata.FradragForEpsSomIkkeHarEPS -> HttpStatusCode.BadRequest.errorJson(
+            "Det er fradrag for EPS, når søker ikke har EPS",
+            "fradrag_for_eps_uten_eps",
+        )
+        KunneIkkeLageGrunnlagsdata.FradragManglerBosituasjon -> HttpStatusCode.BadRequest.errorJson(
+            "Fradrags er ikke innenfor bosituasjonsperioden",
+            "fradragsperiode_utenfor_bosituasjonperiode",
+        )
+        KunneIkkeLageGrunnlagsdata.MåLeggeTilBosituasjonFørFradrag -> HttpStatusCode.BadRequest.errorJson(
```

Comment:
Dette er vel ikke en feil vi kan sende fra oss på dette http-kallet!? En bruker har prøvd å oppdatere bosituasjonen for så å få beskjed at han må leggetil bosituasjon før fradrag? 😬 Tenker det er fair at domenet heller kaster noen exceptions dersom typesystemet ikke kan hjelpe oss her.

Reply:
da må man vel lage en del flere mappinger for at den ikke skal sendes til frontend, vel?